scio-elasticsearch/src/main/java/com/spotify/scio/elasticsearch/ElasticsearchIO.java

```diff
@@ -0,0 +1,228 @@
+    /**
+     * Returns a tranform for writing to Elasticsearch cluster for a given name.
+     *
+     * @param clusterName name of the Elasticsearch cluster
+     */
+    public static Bound withClusterName(String clusterName) {
+      return new Bound().withClusterName(clusterName);
+    }
+
+    /**
+     * Returns a transform for writing to the Elasticsearch cluster for a given servers.
+     *
+     * @param servers endpoints for the Elasticsearch cluster
+     */
+    public static Bound withServers(InetSocketAddress[] servers) {
+      return new Bound().withServers(servers);
+    }
+
+    /**
+     * Returns a transform for writing to Elasticsearch cluster by providing
+     * slight delay specified by flushInterval.
+     *
+     * @param flushInterval delay applied to buffer elements. Defaulted to 1 seconds.
+     */
+    public static Bound withFlushInterval(Duration flushInterval) {
+      return new Bound().withFlushInterval(flushInterval);
+    }
+
+    public static class Bound extends PTransform<PCollection<IndexRequestWrapper>, PDone> {
+      private final String clusterName;
+      private final InetSocketAddress[] servers;
+      private final Duration flushInterval;
+
+      private Bound(final String clusterName,
+                    final InetSocketAddress[] servers,
+                    final Duration flushInterval) {
+        this.clusterName = clusterName;
+        this.servers = servers;
+        this.flushInterval = flushInterval == null? Duration.ofSeconds(1L): flushInterval;
+      }
+
+      Bound() {
+        this(null, null, null);
+      }
+
+      public Bound withClusterName(String clusterName) {
+        return new Bound(clusterName, servers, flushInterval);
+      }
+
+      public Bound withServers(InetSocketAddress[] servers) {
+        return new Bound(clusterName, servers, flushInterval);
+      }
+
+      public Bound withFlushInterval(Duration flushInterval) {
+        return new Bound(clusterName, servers, flushInterval);
+      }
+
+      @Override
+      public PDone expand(final PCollection<IndexRequestWrapper> input) {
+
+        if (clusterName == null) {
+          throw new IllegalStateException(
+              "need to set clustername of ElasticsearchIO.Write transform");
+        }
+
+        if (servers == null) {
+          throw new IllegalStateException(
+              "need to set clustername of ElasticsearchIO.Write transform");
+        }
+
+        input
+            .apply("Assign To Shard", ParDo.of(new AssignToShard()))
+            .apply("Re-Window to Global Window", Window.<KV<Long, IndexRequestWrapper>>into(new GlobalWindows())
+                       .triggering(Repeatedly.forever(
+                           AfterProcessingTime
+                               .pastFirstElementInPane()
+                               .plusDelayOf(javaToJoda(flushInterval))))
+                       .discardingFiredPanes())
+            .apply(GroupByKey.create())
```

Comment:
Not sure if I understand your suggestion but what I am trying to achieve here is to manually create Iterable[T] via sharding and grouping elements assigned to the shard together. They key here is flushInterval which does not exist in default implementation of writer transform which was our problem. Basically without it, this transform writes too fast without batching and too many requests are sent to ES. `numOfWorkers` and rest are arbitrary, as noted above, just trying to create efficient Iterable[T]. Can you elaborate on your suggestion? Thanks!